src/visitable.cpp

```diff
@@ -717,13 +731,13 @@ static long charges_of_internal( const T& self, const itype_id& id, long limit )
 }
 
 template <typename T>
-long visitable<T>::charges_of( const std::string &what, int limit ) const
+long visitable<T>::charges_of( const std::string &what, long limit ) const
 {
     return charges_of_internal( *this, what, limit );
 }
 
 template <>
-long visitable<inventory>::charges_of( const std::string &what, int limit ) const
+long visitable<inventory>::charges_of( const std::string &what, long limit ) const
```

Comment:
Now that I'm back at my laptop I can see that I changed this in 1d064bb358b6e44fc721a783f0d9efb94188fb75 specifically because it was causing an overflow error when I was testing it.
I'l add a commit to shrink everything to int.